### PR TITLE
Added datacommons-bar 'subscribe' event listener to handle date change events from datacommons-slider. 

### PR DESCRIPTION
**Pull request by dwnoble**
- Added datacommons-bar 'subscribe' event listener to handle date change events from datacommons-slider. 
- Updated error display for all components.

## Bar chart slider integration

Example usage:
```
  <datacommons-bar
    apiRoot="http://localhost:8080"
    places="geoId/06 geoId/11 geoId/12"
    date="HIGHEST_COVERAGE"
    title="Life expectancy vs Median age in California, the District of Columbia, and Florida (${date})"
    subscribe="dc-bar"
    variables="LifeExpectancy_Person Median_Age_Person"
  >
    <div slot="footer">
      <datacommons-slider
        apiRoot="http://localhost:8080"
        places="geoId/06 geoId/11 geoId/12"
        publish="dc-bar"
        variables="LifeExpectancy_Person Median_Age_Person"
      ></datacommons-slider>
    </div>
  </datacommons-bar>
```
Screenshot 2024-07-26 at 4 09 48 PM

## Error display update:

MG3mrvpTrZLmP2V
76Vd74zssCVhEcR

### CODE REVIEW

**Review by juliawu — APPROVED** [review]
Very nice! Let's add some examples to the docsite as well of using a slider with a bar chart.

File: static/js/components/tiles/bar_tile.tsx [hunk 1]
```diff
@@ -153,6 +165,24 @@ export function BarTile(props: BarTilePropType): JSX.Element {
   }, [props, barChartData]);
 
   useDrawOnResize(drawFn, chartContainerRef.current);
+
+  /**
+   * Updates the bar tile date when receiving events on the ${props.subscribe}
+   * channel. Used to conenct the datacommons-slider component to this
```
[the remark this answers is not in the document]
Comment (dwnoble) [reply static/js/components/tiles/bar_tile.tsx:171]: fixed

File: static/js/components/tiles/bar_tile.tsx [hunk 2]
```diff
@@ -153,6 +165,24 @@ export function BarTile(props: BarTilePropType): JSX.Element {
...
+   * component
+   */
+  useEffect(() => {
+    if (props.subscribe) {
+      self.addEventListener(
```
[the remark this answers is not in the document]
Comment (dwnoble) [reply static/js/components/tiles/bar_tile.tsx:176]: Good catch! Yes it should be cleaned up- added a function to do that here and to the map_tile.tsx as well

**Comment by dwnoble** [comment]
> Very nice! Let's add some examples to the docsite as well of using a slider with a bar chart.

Will do

**Pull request merged**